src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/stash/StashApiClient.java

```diff
@@ -414,6 +428,94 @@ public String call() throws Exception {
         return response;
     }
 
+    private String putRequest(String path, String status) throws UnsupportedEncodingException {
+        logger.log(Level.FINEST, "PR-PUT-REQUEST:" + path + " with: " + status);
+        HttpClient client = getHttpClient();
+	client.getState().setCredentials(AuthScope.ANY, credentials);
+
+        PutMethod httpput = new PutMethod(path);
+        //http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html; section 14.10.
+        //tells the server that we want it to close the connection when it has sent the response.
+        //address large amount of close_wait sockets client and fin sockets server side
+	httpput.setRequestHeader("Connection", "close");
+        httpput.setRequestHeader("X-Atlassian-Token", "no-check"); //xsrf
+
+         if (status != null) {
+            ObjectNode node = mapper.getNodeFactory().objectNode();
+            node.put("status", status);
+            StringRequestEntity requestEntity = null;
+            try {
+                requestEntity = new StringRequestEntity(
+                        mapper.writeValueAsString(node),
+			"application/json",
+                        "UTF-8");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            httpput.setRequestEntity(requestEntity);
+        }
+
+         String response = "";
+        FutureTask<String> httpTask = null;
+        Thread thread;
+
+         try {
+            //Run the http request in a future task so we have the opportunity
+            //to cancel it if it gets hung up; which is possible if stuck at
+            //socket native layer.  see issue JENKINS-30558
```

Comment:
That issue is closed with "cannot reproduce". https://issues.jenkins-ci.org/browse/JENKINS-30558

Reply:
This ticket is mentioned in few other places should we also remove it entirely?

Reply:
At least don't add it blindly. We can cleanup the existing comments in a separate PR.